### PR TITLE
SPOI-8863 New example for using jmsInput operator for reading from SQS

### PR DESCRIPTION
**Pull request by sanjaypujare**
@amberarrow  pls review and merge if appropriate. Note I am going to close the PR https://github.com/DataTorrent/examples/pull/23 and instead requesting this one to be merged in. I have incorporated review comments from that PR as well as the ActiveMQ example.

### CODE REVIEW

**Comment by amberarrow** [comment]
Running "mvn test' fails with:
2016-10-01 08:43:38,162 [main] DEBUG amazonaws.request handleErrorResponse - Received error response: com.amazonaws.AmazonServiceException: The security token included in the request is invalid. (Service: null; Status Code: 403; Error Code: InvalidClientTokenId; Request ID: b1b3a192-a378-559c-b5c1-f1fb78922d34)
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.578 sec <<< FAILURE!
testApplication(com.example.jmsSqs.ApplicationTest)  Time elapsed: 1.476 sec  <<< ERROR!
com.amazonaws.AmazonServiceException: The security token included in the request is invalid. (Service: AmazonSQS; Status Code: 403; Error Code: InvalidClientTokenId; Request ID: b1b3a192-a378-559c-b5c1-f1fb78922d34)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1389)
.......
-----------------------------------------------
Do we need to mock SQS for the test ? If so, this might be useful:
http://www.schibsted.pl/blog/mocking-amazon-sqs-with-elasticmq/
---------------------------------------------------
There are top-level files sqsExampleCreds.properties and sqsOperatorCreds.properties whose
purpose should be explained.

File: tutorials/jmsSqs/src/test/java/com/example/jmsSqs/ApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    Configuration conf = new Configuration(false);
+    
+    conf.set(SqsApplication.SQSDEV_CREDS_FILENAME_PROPERTY, SQS_OPERATOR_CREDS_FILENAME);
+    
+    // read config values from the properties.xml file
+    conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
+    
+    // one can also set or override values in code as below
+    String pre = "dt.operator.fileOut.prop.";
+    conf.set(   pre + "filePath",        FILE_DIR);
```
Comment (amberarrow) [inline tutorials/jmsSqs/src/test/java/com/example/jmsSqs/ApplicationTest.java]: Odd spacing

File: tutorials/jmsSqs/src/test/java/com/example/jmsSqs/ApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    pre = "dt.operator.sqsIn.prop.";
+    // set the subject here since it is dynamically generated
+    conf.set(   pre + "subject",        currentQueueName);
```
Comment (amberarrow) [inline tutorials/jmsSqs/src/test/java/com/example/jmsSqs/ApplicationTest.java]: ditto

**Comment by sanjaypujare** [comment]
@amberarrow  regarding your first comment, it might be a good idea to use the mock to unit test the example but I am not sure it adds much value for the example. It might be useful for the actual JMS operator test. So for now I will disable the test. Also I will remove those extraneous properties files since they are not used other than this unit test

**Comment by amberarrow** [comment]
As far as I know, all the other examples have a functioning local mode test, so it seems worthwhile to stay consistent if the effort is not too great. It also comes in handy for first cut debugging when developing an application and users would appreciate that.

**Comment by sanjaypujare** [comment]
@amberarrow  I have added the elasticMQ based mock and tested both the unit test and the apa package. ElasticMQ mock is a great idea. Let me know if you want me to squash the commits or make more changes

**Comment by amberarrow** [comment]
I had a question above about the 2 properties files at the top level; can we add a comment either within the files themselves or in the README.md explaining their use/purpose ?

Other than that, it looks good; if you squash, I can merge.

**Comment by sanjaypujare** [comment]
@amberarrow  added comments as requested, squashed commits. Ready for merging....

**Comment by tweise** [comment]
Nice work on the test mock. Users need to debug their own code and having the ability to do that with mocked input/output makes it so much easier.

**Pull request merged**